web/profiles/custom/yalesites_profile/modules/custom/ys_core/ys_core.module

```diff
@@ -99,3 +99,35 @@ function ys_core_user_login_form_submit($form, FormStateInterface $form_state) {
   $url = Url::fromRoute('<front>');
   $form_state->setRedirectUrl($url);
 }
+
+/**
+ * Implements hook_page_preprocess_menu__main().
+ */
+function ys_core_preprocess_menu__main(&$variables) {
+  // If submenu, add the top level link as the first link in the submenu.
+
+  foreach ($variables['items'] as &$menuItem) {
+    // Skip menu items that do not have subitems.
+    if (empty($menuItem['below'])) {
+      continue;
+    }
+
+    // Clone top level menu item.
+    $clonedMenuItem = $menuItem;
+
+    // We don't want any other subitems, just the top level.
+    $clonedMenuItem['below'] = NULL;
```

Comment:
Good that we re-reviewed this. There was a small bug where it was cloning ALL subitems into the menu. Since we only want the top level item, I set all the below items to `NULL`.

Reply:
Smart